### PR TITLE
Fix switch kernel canceling

### LINKED ISSUE #604: Notebook Broken if Cancelled "Switch kernel"
sbroken

Note that the default kernel shown in the switched thing is wrong, it shows plos as selected instead of Python 3. And cancelling,  break the notebook with the switch kernel not working.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #604.

### CODE REVIEW

File: src/docregistry/kernelselector.ts [hunk 1]
```diff
@@ -106,7 +106,9 @@ function selectKernelForContext(context: IDocumentContext<IDocumentModel>, host?
     };
     return selectKernel(options);
   }).then(kernel => {
-    context.changeKernel(kernel);
+    if (kernel !== void 0) {
```
Comment (afshin) [inline src/docregistry/kernelselector.ts]: Maybe just `if (kernel) {`...

File: src/notebook/notebook/default-toolbar.ts [hunk 2]
```diff
@@ -412,8 +412,13 @@ class KernelIndicator extends Widget {
       this.node.title = 'No Kernel!';
     }
     panel.kernelChanged.connect((c, kernel) => {
-      this._handleStatus(kernel, kernel.status);
-      kernel.statusChanged.connect(this._handleStatus, this);
+      if (kernel) {
```
Comment (afshin) [inline src/notebook/notebook/default-toolbar.ts]: Like here 😄

File: test/src/docmanager/mockcontext.ts [hunk 3]
```diff
@@ -82,7 +82,11 @@ class MockContext<T extends IDocumentModel> implements IDocumentContext<T> {
   }
 
   changeKernel(options: IKernel.IModel): Promise<IKernel> {
-    this._kernel = new MockKernel(options);
+    if (options === void 0) {
```
Comment (afshin) [inline test/src/docmanager/mockcontext.ts]: Same sort of thing, `if (options) {`... etc.

**Comment by blink1073** [comment]
Updated.

**Comment by afshin** [comment]
👍

**Pull request merged**